src/common/pipes/CustomParseDatePipe.ts

```diff
@@ -0,0 +1,12 @@
+import { PipeTransform, Injectable, BadRequestException } from '@nestjs/common';
+
+@Injectable()
+export class CustomParseDatePipe implements PipeTransform<string, Date> {
+  transform(value: string): Date {
+    const date = new Date(value);
+    if (isNaN(date.getTime())) {
+      throw new BadRequestException('Invalid date format');
+    }
+    return date;
```

Comment:
Maybe wrap this with a try catch, just in case some input breaks new Date() expected param
```suggestion
    try {
      const date = new Date(value);
      if (isNaN(date.getTime())) {
        throw new BadRequestException('Invalid date format');
      }
      return date;
    } catch (error) { ... }
```

Reply:
A atribuição `const date = new Date(value);` sempre retornará um date, mesmo para strings teoricamente inválidas. Por isso, não será lançada uma exceção.

Reply:
Tens toda razão, não sabia disso, obrigado